cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceRuntimeService.java

```diff
@@ -482,10 +481,8 @@ public Void call() throws Exception {
           context.setState(new ProgramState(ProgramStatus.INITIALIZING, null));
           if (mapReduce instanceof ProgramLifecycle) {
```

Comment:
Yes. But `MapReduce` interface does not extend the `ProgramLifeCycle` but `AbstractMapReduce` does.